packages/zowe-explorer-api/src/profiles/ProfilesCache.ts

```diff
@@ -539,11 +539,10 @@ export class ProfilesCache {
     }
 
     private shouldRemoveTokenFromProfile(profile: zowe.imperative.IProfileLoaded, baseProfile: zowe.imperative.IProfileLoaded): boolean {
-        return (baseProfile?.profile?.host &&
-            baseProfile?.profile?.port &&
+        return ((baseProfile?.profile?.host || baseProfile?.profile?.port) &&
             profile?.profile?.host &&
             profile?.profile?.port &&
-            (baseProfile?.profile.host !== profile?.profile.host || baseProfile?.profile.port !== profile?.profile.port) &&
+            (baseProfile?.profile.host !== profile?.profile.host || (profile?.profile.user && profile?.profile.password)) &&
```

Comment:
Also, it might be good to keep the `port` check as well 😋

Reply:
Agreed, it might be good to keep the `port` check here as well as on line 542